src/tls.c

```diff
@@ -904,8 +913,8 @@ static int Hmac_OuterHash(Hmac* hmac, unsigned char* mac)
                 (word32)digestSz);
         if (ret == 0)
             ret = wc_HashFinal(&hash, hashType, mac);
-        wc_HashFree(&hash, hashType);
     }
+    wc_HashFree(&hash, hashType);
```

Comment:
'hash' may not have been initialised properly.
It' looks fine with the implementation but don't rely on that.